src/Database/PostgreSQL/Consumers/Components.hs

```diff
@@ -269,7 +273,7 @@ spawnDispatcher ConsumerConfig{..} cs cid semaphore
 
       return (batchSize > 0)
 
-    reserveJobs :: Int -> m ([job], Int)
+    reserveJobs :: Int -> m (Either job [job], Int)
```

Comment:
The Either is quite confusing. Could you add a comment about the idea behind it so that people don't have to figure out what it's supposed to mean? I wonder if a simple data type wouldn't be better here.

Reply:
And by "better" I mean more readable...